private/with-npm-packages/README.md

```diff
@@ -0,0 +1,116 @@
+
+``` bash
+export TOOL_NODE_FLAGS="--max_old_space_size=4096" # can be needed on secenarios with many files
+```
+
+## Experiment
+
+The experiment consists in several in testing the rebuild performance on different file scenarios for the app level and packages.
+
+The tests have been performed on the next computer specs:
+
+``` shell
+MacBook Pro (Retina, 15-inch, Mid 2015)
+Processor 2.8 GHz Intel Core i7 4 cores
+Memory 16 GB 1600 MHz DDR3
+SSD
+```
+
+The results are available on `logs/` folder, but the next table sums it up and serve for the analysis.
+
+| n  | App files | Package #A files | Package #B files | Change on | Total |
+|----|-----------|------------------|------------------|-----------|-------|
+| 1  |    1000   |         1        |         1        |    app    | 2543  |
+| 2  |    5000   |         1        |         1        |    app    | 6519  |
+| 3  |    1000   |       5000       |         1        |    app    | 3405  |
+| 4  |    1000   |       5000       |         1        |    p#A    | 2091  |
+| 5  |    5000   |       1000       |         1        |    app    | 6602  |
+| 6  |    5000   |       1000       |         1        |    p#A    | 3955  |
+| 7  |    1000   |       5000       |       1000       |    app    | 3547  |
+| 8  |    1000   |       5000       |       1000       |    p#A    | 1569  |
+| 9  |    1000   |       5000       |       1000       |    p#B    | 1528  |
+| 10 |    1000   |       5000       |       5000       |    app    | 4547  |
+| 11 |    1000   |       5000       |       5000       |    p#A    | 3365  |
+| 12 |    1000   |       5000       |       5000       |    p#B    | 3127  |
+
+> Note: The times showed in the table above correspond to the times on code change with meteor cached.
+
+### Analysis
+
+After the previous results on the experiment we can clearly see that:
+
+- Changing on package is faster than doing it on app level. The reason for that is that with babel compilation we have gained paralelism in package compilation and transpilation on modules on demand. Meteor only has to put together the app modules and NPM dependencies. (See `3` and `4`, and `10`, `11` and `12`).
+
+In comparision with [the previous experiment](../with-meteor-packages/README.md#Analysis), we can clearly see that:
+
+- Meteor apps are better optimized than meteor packages, but meteor builder is a bit outdated in terms of project construction. For the moment, we can only ensure that having NPM packages and an exernal way to do ES6 transpilation for those, is better than having Meteor to handle in its own means.
+
+As always, during the experiment I also noted that:
+
+- Having large amount of files and links between them is critical on app level and makes the initial load of the builder (timings without the cache) to get so much time to rebuild (many many minutes! insane!).
```

Comment:
what do you mean by "links between them"? just imports/exports? how many files do we need to have "many minutes"? it also occurs with this "npm packages" strategy?

Reply:
> what do you mean by "links between them"? just imports/exports?

Yes, imports/exports, I wrote like this because is important the distinction as the Meteor builder work. Having large amount of files without being imported affects only to the `prepareProjectToBuild`, which is not affecting so much, while having them "linked" affects significatively on the "BuildApp" phase, getting many amount of minutes on the first cache creation.

> how many files do we need to have "many minutes"? it also occurs with this "npm packages" strategy?

I got that on performing the "3" scenario, with 5000 files on app level, and with empty cache. The thing is that once that finished, other scenarios went faster and not as critical. So clearing the cache adds the issue.